extensions/openshift-client/runtime/src/main/java/io/quarkus/it/openshift/client/runtime/OpenShiftClientProducer.java

```diff
@@ -0,0 +1,26 @@
+package io.quarkus.it.openshift.client.runtime;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.openshift.client.DefaultOpenShiftClient;
+import io.fabric8.openshift.client.OpenShiftClient;
+import io.quarkus.arc.DefaultBean;
+
+@Singleton
+public class OpenShiftClientProducer {
+
+    @DefaultBean
+    @Singleton
+    @Produces
+    public OpenShiftClient openshiftClient(Config config) {
+        return new DefaultOpenShiftClient(config);
+    }
+
+    @PreDestroy
+    public void destroy(OpenShiftClient client) {
```

Comment:
This needs to be 
```java
 public void destroy(@Observes @Destroyed OpenShiftClient client) {
...
}
```

AFAIK

Reply:
I could not get it to work. `@Destroyed` requires a scope.
I could not find something that looked like what you are suggesting.
do you have another idea, or should I revert the code?

Reply:
this works:
```
    public void destroy(@Observes @Destroyed(ApplicationScoped.class) Object container, OpenShiftClient client) {
        if (client != null) {
            LOGGER.info("closing openshift client");
            client.close();
        }
    }
```

I did not know if the client could be null, so I implemented defensively. 
I squashed the commits.